### PR TITLE
Enable docker compose

### PR DESCRIPTION
**Pull request by electrocucaracha**
Utilizing `docker-compose` tool can reduce the number of instructions used to deploy the cloud provider service.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,11 +49,8 @@ Another alternative is to run it as a container, but this will require to mount
 the docker socket inside the container:
 
 ```sh
-docker build . -t aojea/cloud-provider-kind:v0.1
-# using the host network
-docker run --rm --network host -v /var/run/docker.sock:/var/run/docker.sock aojea/cloud-provider-kind:v0.1
-# or the kind network
-docker run --rm --network kind  -v /var/run/docker.sock:/var/run/docker.sock aojea/cloud-provider-kind:v0.1
+# using the `kind` network (`host` is the default value for NET_MODE)
+NET_MODE=kind docker compose up -d
```
Comment (BenTheElder) [inline README.md:53]: I feel like we should document how to do this without compose still.

E.G. if we start to publish a container image that is pre-built (when the project is more mature) then you will want to skip building most of the time and we'll want this simple docker run step again (or binary run on the host)

We're also going to want to figure out e.g. podman, where you won't be mounting the socket necessarily.

**Comment by aojea** [comment]
hi Victor, thanks for the PR but as Ben say we have a considerable use base of podman users ... and now nerdctl

I would rather prefer to make the deployment containerized ... but in a standard way that serves all ...  but I don't know if this will work also in Mac and Windows, since we need access to the host

**Comment by aojea** [comment]
I don't mind adding it as an additional subsection or alternative way, but I prefer to keep the existing ... we may also rename the iamge to not use my github and use something more generic

**Comment by aojea** [comment]
/lgtm
/approve

Gracias Victor

**Pull request merged**